src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs

```diff
@@ -0,0 +1,158 @@
+
+[Generator(LanguageNames.CSharp)]
+class TextColorToGenerator : IIncrementalGenerator
+{
+	const string iTextStyleInterface = "Microsoft.Maui.ITextStyle";
+	const string iAnimatableInterface = "Microsoft.Maui.Controls.IAnimatable";
+	const string mauiControlsAssembly = "Microsoft.Maui.Controls";
+
+	public void Initialize(IncrementalGeneratorInitializationContext context)
+	{
+		// Get All Classes in User Library
+		var userGeneratedClassesProvider = context.SyntaxProvider.CreateSyntaxProvider(
+			static (syntaxNode, cancellationToken) => syntaxNode is ClassDeclarationSyntax { BaseList: not null },
+			static (context, cancellationToken) => (ClassDeclarationSyntax)context.Node);
+
+		// Get Microsoft.Maui.Controls Assymbly Symbol
+		var mauiControlsAssemblySymbolProvider = context.CompilationProvider.Select(
+			static (compilation, token) => compilation.SourceModule.ReferencedAssemblySymbols.Single(q => q.Name == mauiControlsAssembly));
+
+		var inputs = userGeneratedClassesProvider.Collect()
+						.Combine(mauiControlsAssemblySymbolProvider)
+						.Select(static (combined, cancellationToken) => (UserGeneratedClassesProvider: combined.Left, MauiControlsAssemblySymbolProvider: combined.Right))
+						.Combine(context.CompilationProvider)
+						.Select(static (combined, cancellationToken) => (combined.Left.UserGeneratedClassesProvider, combined.Left.MauiControlsAssemblySymbolProvider, Compilation: combined.Right));
+
+		context.RegisterSourceOutput(inputs, (context, collectedValues) =>
+		Execute(context, collectedValues.Compilation, collectedValues.UserGeneratedClassesProvider, collectedValues.MauiControlsAssemblySymbolProvider));
+	}
+
+	static void Execute(SourceProductionContext context, Compilation compilation, ImmutableArray<ClassDeclarationSyntax> userGeneratedClassesProvider, IAssemblySymbol mauiControlsAssemblySymbolProvider)
+	{
+		var textStyleSymbol = compilation.GetTypeByMetadataName(iTextStyleInterface);
+		var iAnimatableSymbol = compilation.GetTypeByMetadataName(iAnimatableInterface);
+
+		if (textStyleSymbol is null || iAnimatableSymbol is null)
+			return;
+
+		var textStyleClassList = new List<(string ClassName, string ClassAcessModifier, string Namespace)>();
+
+		// Collect Microsoft.Maui.Controls that Implement ITextStyle
+		var mauiTextStyleImplementors = mauiControlsAssemblySymbolProvider.GlobalNamespace.GetNamedTypeSymbols().Where(x => x.AllInterfaces.Contains(textStyleSymbol, SymbolEqualityComparer.Default)
+				&& x.AllInterfaces.Contains(iAnimatableSymbol, SymbolEqualityComparer.Default));
+
+		foreach (var namedTypeSymbol in mauiTextStyleImplementors)
+			textStyleClassList.Add((namedTypeSymbol.Name, "public", namedTypeSymbol.ContainingNamespace.ToDisplayString()));
+
+		// Collect All Classes in User Library that Implement ITextStyle
+		foreach (var classDeclarationSyntax in userGeneratedClassesProvider)
+		{
+			var declarationSymbol = compilation.GetSymbol<INamedTypeSymbol>(classDeclarationSyntax);
```

Comment:
I'm not sure if the semantic model can return an `IErrorTypeSymbol`, for which this call will crash due to invalid cast.

Reply:
Why this will be an `IErrorTypeSymbol`?

Reply:
Because I think nothing guarantees otherwise. There could be some compile errors that causes roslyn to not have an INamedTypeSymbol for this node. It's safer to keep it an `as TSymbol` instead of `(TSymbol?)`

Reply:
I would say that we want this to fail. If the type isn't the wanted we should see an exception

Reply:
@pictos It's not a good experience to see an exception from the generator while typing erroneous code in the IDE.